src/ui/App.js

```diff
@@ -1,47 +1,85 @@
 import Api from "./Api.js";
+import { Deck, DeckExpiredError } from "./Deck.js";
 import Hand from "./Hand.js";
-import { useState, useCallback } from "react";
+import { useState, useCallback, useEffect } from "react";
 
 /**
    Note: For each game, we'll only allow the "Draw" to happen once.
    This means we should hide the "Draw" button after it is clicked.
    Better yet, let's change it to a different "Play Again" button that
    resets everything and plays another hand (with a new Deck).
 **/
-export default function App({ initialCards }) {
+export default function App({ initialCards, initialDeck, initialEnded }) {
   const [cards, setCards] = useState(initialCards);
   const [selected, setSelected] = useState([]);
+  const [deck, setDeck] = useState(initialDeck);
+  const [dealt, setDealt] = useState(initialEnded);
+  const [error, setError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState("");
+  console.log("initialDeck: " + initialDeck);
+  console.log("current deck: " + deck);
+  console.log("setDeck? " + setDeck);
 
-  function toggleSelected(index) {
-    if (!selected.includes(index)) {
-      setSelected(selected.concat([index]));
-    } else {
-      setSelected(selected.filter((elt) => elt !== index));
+  useEffect(() => {
+    if (error) {
+      // restart from scratch
+      let d = new Deck();
+      d.init().then((result) => {
+        d.setSessionStorage();
+        setCards(result);
+        setDeck(d);
+        setError(false);
+        setSelected([]);
+        setDealt(false);
+      });
     }
-  }
+  }, [error]);
+
+  useEffect(() => {
+    if (!error) {
+      console.log("storing deck: " + deck.deckId);
+      console.log(deck.expiry);
+      deck.setSessionStorage();
+      sessionStorage.setItem("hand", JSON.stringify(cards));
```

Comment:
Ah! `sessionStorage` is a global variable (a DOM API I guess?). I was pretty confused because I had never used it before, so I had no idea what it was. :D

Generally, for global DOM APIs, it's a good idea to put them behind method calls if possible.